va/va_vpp.h

```diff
@@ -286,22 +286,152 @@ typedef enum _VAProcColorBalanceType {
     VAProcColorBalanceCount
 } VAProcColorBalanceType;
 
-/** \brief Color standard types. */
+/** Color range types. */
+typedef enum _VAProcColorRange {
+    /** Unknown/unspecified range.
+     *
+     * This will use limited range for YUV and full range for RGB.
+     */
+    VAProcColorRangeUnknown = 0,
+    /** Limited range.
+     *
+     * Used by most video YUV standards, and for RGB in some video contexts.
+     */
+    VAProcColorRangeLimited,
+    /** Full range.
+     *
+     * Used for JPEG images in YUV, and for RGB in all non-video contexts.
+     */
+    VAProcColorRangeFull,
+} VAProcColorRange;
+
+/** Chroma sample location types.
+ *
+ * Only used for YUV 4:2:0 subsampled images.  These values correspond to
+ * the H.264/H.265 syntax element chroma_sample_loc_type_top_field.
+ *
+ */
+typedef enum _VAProcChromaSampleLocation {
+    /** Left.
+     *
+     * This is the default for most video standards (MPEG-2, H.264, H.265),
+     * but typically these can also support other locations.
+     */
+    VAProcChromaSampleLocationLeft = 0,
+    /** Center.
+     *
+     * This is the only option for JPEG and some older video standards (MPEG-1,
+     * H.263).
+     */
+    VAProcChromaSampleLocationCenter = 1,
+    /** Top-left. */
+    VAProcChromaSampleLocationTopLeft = 2,
+    /** Top. */
+    VAProcChromaSampleLocationTop = 3,
+    /** Bottom-left. */
+    VAProcChromaSampleLocationBottomLeft = 4,
+    /** Bottom. */
+    VAProcChromaSampleLocationBottom = 5,
+} VAProcChromaSampleLocation;
```

Comment:
@xfengcarl do you have any comment for this PR,  you have a similar patch for chroma sample location.

Reply:
@xfengcarl what's your comment?

Reply:
there are same definition in PR71. check it.

Reply:
I'd be fine with adopting the same flag structure for chroma siting and full/limited range as there if preferred.  Unclear whether it should still be inside the ColorProperties structure, though - are the flags going to be used anywhere else?  (Currently you have `input_surface_flag` and `output_surface_flag` in the same roles as `surface_color_properties` and `output_color_properties` above.)

Reply:
you will integrate the color properties together, so you want to remove input_surface_flag & output_surface_flag from the PR71?

Reply:
In xfengcarl's code, the bit fields in input_surface_flag and output_surface_flag are defined as following.
bit[31] - encrypted or not
bit[16] - full range or reduced range
bit[1~0] - vertical chromasiting
bit[3~2] - horizontal chromasiting

What about using the remaining bit fields in input_surface_flag and output_surface_flag for colour_primaries,  transfer_characteristics and matrix_coefficients. Below are examples (Assume 3-bit is enough for each one).
bit[6~4] - colour_primaries
bit[9~7] - transfer_characteristics
bit[12~10] - matrix_coefficients

Reply:
@jackie-guo  Ideally they should be 8 bits to match use in MPEG-2 (uimsbf(8) in sequence display extension), H.264 (u(8) in VUI) and H.265 (also u(8) in VUI).  They could be smaller at the moment (largest current value is 22), but that might cause problems later as future versions of the standard specify more values.

While that just about fits if we cram them in to the surface flags, I think it would be cleaner for them to be standalone uint8_t values.  (A bitfield setup could also work?)

@xfengcarl  I think so?  If `input_surface_flag` and `output_surface_flag` are going to contain other things unrelated to colour then do keep them, but I would move the colour properties to the new structure so that they are together.

Reply:
Although some ITU standards have defined colour_primaries/transfer_characteristics/matrix_coefficients, but some other ISO standards or some company standards have not defined colour_primaries/transfer_characteristics/matrix_coefficients. So current driver only takes color range and chromasiting into consideration, does not take colour_primaries/transfer_characteristics/matrix_coefficients into consideration.

Reply:
that's a wrong approach IMO, the driver should take the MAXIMUM common denominator and then proactively fill in the information that is missing (if needed), not exclude necessary fields from its API

also what are the ISO standards that you refer to lacking these properties? they are usually pretty accurate

Reply:
@fhvwy do you have concerns about using union instead of structure?

Reply:
I add a  structure in #71 
typedef struct _VAProcColorProperties {
    /** Chroma sample location.\c VA_CHROMA_SITING_VERTICAL_XXX | VA_CHROMA_SITING_HORIZONTAL_XXX */
    uint8_t chroma_sample_location;
    /** Chroma sample location. \c VA_SOURCE_RANGE_XXX*/
    uint8_t color_range;
    uint8_t reserved[6];
} VAProcColorProperties;
and there are 6 reserved bytes to be used , you can combine your pr with mine smoothly
thanks

Reply:
Thanks for the new definition. The solution looks fine to me.

Reply:
@xfengcarl  Yep, that looks good to me.  I'll try to review your branch again soon, and then rebase this on top of it.